lib/ansible/module_utils/basic.py

```diff
@@ -479,9 +479,11 @@ def tmpdir(self):
             if basedir is not None and not os.path.exists(basedir):
                 try:
                     os.makedirs(basedir, mode=0o700)
-                except (OSError, IOError) as e:
-                    self.warn("Unable to use %s as temporary directory, "
-                              "failing back to system: %s" % (basedir, to_native(e)))
+                except OSError as ex:
+                    self.warn(
+                        f"Unable to use {basedir!r} as temporary directory, "
+                        f"falling back to system: {to_native(ex)!r}"
+                    )
```

Comment:
We now have `error_as_warning` in module_utils, so we can use this:

```suggestion
                except OSError as ex:
                    self.error_as_warning(f"Unable to use {basedir!r} as temporary directory, falling back to system default.", exception=ex)
```